napari/layers/surface/surface.py

```diff
@@ -282,14 +289,37 @@ def gamma(self, value):
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
-        not_disp = list(self.dims.not_displayed)
-        disp = list(self.dims.displayed)
-        indices = np.array(self.dims.indices)
+        # Take vertex_values dimensionality into account if more than one value
+        # is provided per vertex.
+        if self.vertex_values.ndim > 1:
+            values = self.vertex_values[
+                self.dims.indices[: -self.vertices.shape[1]]
+            ]
+            if values.ndim > 1:
+                # If not all dimensions corresponding to the vertex_values
+                # are being sliced through, average over any remaining ones
+                # so that there is always only value per vertex.
+                values = values.mean(axis=tuple(range(0, values.ndim - 1)))
+            self._view_vertex_values = values
+            # Determine which axes of the vertices data are being displayed
+            # and not displayed, but ignoring the additional dimensions
+            # corresponding to the vertex_values.
+            indices = np.array(self.dims.indices[-self.vertices.shape[1] :])
```

Comment:
Again, the `.shape[i]` syntax is a bit confusing. My suggestion is to replace these values with letters at the top of the function: `N, D = self.vertices.shape`. (or similar) Then use those names rather than indexing into a shape.

Reply:
done